### DISCUSSION TITLE
Pressure control with temperature drop

### DISCUSSION THREAD

**Discussion by dad616610** [e2nIEE/pandapipes#624] [Q&A] [answered]
Hello!

I was reading a documentation about [Pressure Control](https://pandapipes.readthedocs.io/en/v0.8.5/components/press_control/press_control_component.html) and got confused by the note there:

> The temperature at the inlet and outlet junction will not be adapted by the pressure control unit. Therefore, these components usually operate isothermal (inlet temperature = outlet temperature). It is assumed that temperature changes due to compression or expansion (Joule-Thomson-effect) are balanced internally by adding or removing heat.

The first sentence says, that the temperature at the outlet will not be adapted, but the last sentence says, that the temperature changes will be balanced internally. Am I misunderstanding the last sentence? Does it mean, that the user will inject the temperature drop "manually" via the `qext_w` parameter? Is this the only way for now?

Here is a minimal example with pressure control with no temperature drop and with manual temperature drop insertion in the pipe, after the pressure control element
<details><summary>Example</summary>
<p>

```py
import pandapipes as pp


def main():
    net = pp.create_empty_network("mynetwork", fluid="lgas")

    pp.create_junctions(net, 4, 1, 300)
    pp.create_ext_grid(net, junction=0, p_bar=5, t_k=300)

    pp.create_pressure_control(
        net,
        from_junction=1,
        to_junction=2,
        controlled_junction=2,
        controlled_p_bar=0.5,
    )
    pp.create_sink(net, junction=3, mdot_kg_per_s=6)

    pp.create_pipes_from_parameters(
        net, from_junctions=[0, 2], to_junctions=[1, 3], length_km=0.1, diameter_m=0.2
    )

    # uncomment to get temp drop
    # net.pipe.loc[1, "qext_w"] = 2e4

    pp.pipeflow(net, mode="all", friction_model="colebrook", iter=10000)

    print(net.res_press_control)
    print(net.res_pipe)


if __name__ == "__main__":
    main()
```

</p>
</details> 

P.S. the "latest" documentation doesn't render function's description, have to use 0.8.5 version
<details><summary>Screenshot with unrendered function description on "latest" version</summary>
<p>

![image](https://github.com/e2nIEE/pandapipes/assets/147147326/6f11260f-3bc9-421f-9ce4-a3d4fce6b122)

</p>
</details>

**Comment by dlohmeier** [comment, marked as answer]
Hello @dad616610 ,

thanks for pointing at the documentation problem, we will have a look at it.
Regarding the question on the pressure controller, here's what happens: The temperature at the outlet node is just whatever tfluid_k you give it, the pressure controller will not have any influence on it. When performing a sequential hydraulics and heat transfer simulation (mode "all"), the temperature that occurs before the pressure controller will propagate through it, without any changes. Yet, in a real throttle valve, you get a temperature drop due to expansion. Would you like to model such behavior? In that case, you would need to perform several simulations to correctly model the influence of temperature and pressure on other gas properties as density (we call this bi-directional coupling and it's currently WIP on a feature branch of pandapipes). Still, the model of the pressure controller (in your case, it should be a throttle) does not implement any such temperature change, so you would have to create your own adapted pressure control component or an external controller that implements this behavior (e.g. by adding a heat_exchanger with pre-calculated q_ext). 

I hope that answers your question. Please have a look at the pandapipes and pandapower control docs (https://pandapower.readthedocs.io/en/latest/control.html) if you want to learn more about how to create such a controller. 

Best regards!
